### DISCUSSION TITLE
Azure Keyvault data for On Prem Vsphere-Iso packer builds

### DISCUSSION THREAD

**Discussion by Morgarnus** [vmware/packer-examples-for-vsphere#169] [Help] [answered]
How can I  bring a key from Azure Keyvault to use in packer for Vsphere-Iso Builds?

And what is the best practice for using packer on On Prem VSphere setup?

**Comment by tenthirtyam** [comment, marked as answer]
Probably best to present these questions on the discussion forums for Packer, which can be found in the README.md at GitHub.com/hashicorp/packer.

Ryan

**Reply by Morgarnus** [reply]
Thanks,    Just for anyone else who may look for this. This is how I have done it and it works ok for me, I am sure there will be a better way but this does work.

Use a Powershell script to pull down Azure Variable

```powershell
$env:test = Get-AzKeyVaultSecret -VaultName "example" -Name "example name in vault"
```
Then inside packer variables it can be used like so

```hcl
variable "Sample_key" {
   type      = string
   default  = "${env("test")}"
   sensetive = true
}
```
hope this helps

Thanks for pointing me in correct direction.
